test/services/validator_check_active_service_test.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class ValidatorCheckActiveWorkerTest < ActiveSupport::TestCase
+  setup do
+    create(:epoch_wall_clock, epoch: 123)
+  end
+
+  test 'validator with active stake and nondelinquent should be active' do
+    v = create(:validator, :with_score, account: 'account1')
+    create(:validator_history, account: 'account1')
+    create(:validator_block_history, validator: v, epoch: 122)
+
+    ValidatorCheckActiveService.new.update_validator_activity
+
+    assert_equal true, Validator.find_by(account: 'account1').is_active
+  end
+
+  test 'validator with zero active stake should be inactive' do
+    v = create(:validator, :with_score, account: 'account2')
+    create(:validator_history, account: 'account2', active_stake: 0)
+    create(:validator_block_history, validator: v, epoch: 122)
+
+    ValidatorCheckActiveService.new.update_validator_activity
+
+    assert_equal false, Validator.find_by(account: 'account2').is_active
```

Comment:
```suggestion
    refute v.reload.is_active
```
Change accordingly in other tests.